app/models/submission.rb

```diff
@@ -90,4 +90,8 @@ def automated_feedback
     feedback << exercise_representation if has_automated_feedback?
     feedback
   end
+
+  def viewable_by?(user)
+    solution.mentors.include?(user)
```

Comment:
I'm pretty unsure about this, but leaving this up to you @iHiD. We might want to make sure that only the iterations the mentor is mentoring is viewable.

Reply:
Yeah, this makes sense. It should also allow the user who's solution it is. Please add a test.

With this whole UI, there's going to be a student-equivelent that's almost identical.